### DISCUSSION TITLE
is it possible to use the name of the folder that helmfile is in as the namespace of the releases?

### DISCUSSION THREAD

**Discussion by yidongw** [helmfile/helmfile#879] [Questions]
For example, if I have a `helmfile.yaml` in `helmfiles/traefik` folder.

Inside the `helmfiles/traefik/helmfile.yaml`, I have a list of releases.
Is it possible to use `traefik` for the namespace of all the releases listed in `helmfiles/traefik/helmfile.yaml`?

**Comment by yxxhero** [comment]
@yidongw first of all. you can get the name of the folder by template functions. you can find you want in https://helmfile.readthedocs.io/en/latest/templating_funcs/ and [sprig.](https://github.com/Masterminds/sprig)

**Comment by yidongw** [comment]
Thank you. to close the loop here.
I can write release with
```yaml
    namespace: {{ exec "../../scripts/get-namespace.sh" list }}
```
and in the get-namespace.sh
```bash
#!/bin/bash

# Get the full path of the current directory
current_dir=$(pwd)

# Extract the last part of the path (the current folder name)
folder_name=$(basename $current_dir)

echo ${folder_name}
```
